cogent/src/Cogent/DataLayout/Core.hs

```diff
@@ -0,0 +1,91 @@
+{-# LANGUAGE DeriveFunctor #-}
+module Cogent.DataLayout.Core where
+  
+import Data.Map (Map)
+
+import Text.Parsec.Pos (SourcePos)
+
+import Cogent.Common.Syntax (TagName, FieldName)
+
+
+{- CORE DATALAYOUT TYPES -}
+
+-- A range of bit indices into a data type.
+--
+-- Should satisfy `bitSizeBR >= 1` and `bitOffset >= 0`.
```

Comment:
With natural number types, my preference is for newtype over type synonym, because it prevents accidental assignment of non-natural numbers to natural ones. Can use `{-# LANGUAGE NewtypeDeriving #-}` to inherit the Num, Ord, Eq from Integer. Except it seems like there is a preference for type synonyms over new type elsewhere in the compiler. What are your thoughts?

Reply:
I was thinkinf of things like http://hackage.haskell.org/package/ghc-prim-0.5.2.0/docs/GHC-Types.html#t:Word